arch/sim/src/sim/up_oneshot.c

```diff
@@ -384,6 +450,19 @@ void up_timer_update(void)
 
   FAR sq_entry_t *entry;
 
+#ifdef CONFIG_SIM_ONESHOT_HOST_TIMER
+
+  /* If we are simulating the 'tick' events from the Idle Task
+   * we return here as we are currently using the oneshot host
+   * timer.
+   */
+
+  if (up_interrupt_context() == false)
+    {
+      return;
+    }
+#endif
+
   clock_timespec_add(&g_current, &tick, &g_current);
```

Comment:
It's better to add a new static funtion like this:
```
clock_timespec_add(&g_current, &tick, &g_current);
for (entry = sq_peek(&g_oneshot_list); entry; entry = sq_next(entry))
    {
      sim_process_tick(entry);
    }
```
and let's up_timer_update and host_sigalrm_handler call it.

Reply:
That's a good idea I will do this

Reply:
Yes, it's wrong to call up_interrupt_context here to check alarm signal because the new signal handler(actually, SIGUSR1 is used for SMP now) may add later.